sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/model/BindableQuery.kt

```diff
@@ -148,6 +151,22 @@ abstract class BindableQuery(
           }
       ))
     }
+
+    // If we'd previously assigned a type to this argument other than NULL, and later encounter NULL,
+    // we should update the existing type to be nullable.
+    // i.e. WHERE (data = :foo OR :foo IS NULL)
+    if (bindArg.argumentType().sqliteType == NULL && current.type.sqliteType != NULL) {
```

Comment:
can this if/else just compute what `base` should be and then we can move all the code out that does 

```
remove(current)
add(current.copy(
  index = index ?: current.index,
  type = base.type.run {
    ...
  }
```

Reply:
Yeah, avoid the duplication a bit? Sounds good.